sklearn/model_selection/validate.py

```diff
@@ -0,0 +1,905 @@
+"""
```

Comment:
Do I preserve the `master`'s order?

I ordered it this way... 
```
function_a
_helper_used_first_in_a
function_b
_helper_used_first_in_b
```
So it becomes easier to find the helpers :)